src/QMCDrivers/DMC/WalkerControlMPI.cpp

```diff
@@ -239,7 +264,9 @@ void WalkerControlMPI::determineNewWalkerPopulation(int Cur_pop,
 void WalkerControlMPI::swapWalkersSimple(MCWalkerConfiguration& W)
 {
   std::vector<int> minus, plus;
-  determineNewWalkerPopulation(Cur_pop, num_contexts_, MyContext, NumPerNode, FairOffSet, minus, plus);
+  //legacy code does not modify NumPerNode in this call so we copy NumPerNode
+  std::vector<int> num_per_node(NumPerNode);
+  determineNewWalkerPopulation(Cur_pop, num_contexts_, MyContext, num_per_node, FairOffSet, minus, plus);
```

Comment:
If I read the code correctly, the initial value of num_per_node seems get used but you passing an array of zero.